include/deal.II/meshworker/mesh_loop.h

```diff
@@ -392,6 +392,116 @@ namespace MeshWorker
                     queue_length,
                     chunk_size);
   }
+
+  /**
+   * This is a variant of the mesh_loop function, that can be used for worker
+   * and copier functions that are member functions of a class.
+   *
+   * The argument passed as @p end must be convertible to the same type as @p
+   * begin, but doesn't have to be of the same type itself. This allows to
+   * write code like <code>mesh_loop(dof_handler.begin_active(),
+   * dof_handler.end(), ...</code> where the first is of type
+   * DoFHandler::active_cell_iterator whereas the second is of type
+   * DoFHandler::raw_cell_iterator.
+   *
+   * The @p queue_length argument indicates the number of items that can be
+   * live at any given time. Each item consists of @p chunk_size elements of
+   * the input stream that will be worked on by the worker and copier
+   * functions one after the other on the same thread.
+   *
```

Comment:
I think that our less experienced users sometimes struggle with the syntax of these sorts of functions (I know that I certainly did when first learning to use the workstream functions). Could we put an small example showing how to use this? From your test:

```
template <int dim, int spacedim>
class TestClass
{
public:
  void
  cell_worker(const CellIteratorType &cell, ScratchData &, CopyData &);

  ...
};

  ...
  TestClass<dim, spacedim> test_class;
  ScratchData              scratch;
  CopyData                 copy;

  mesh_loop(tria.begin_active(),
            tria.end(),
            test_class,
            &TestClass<dim, spacedim>::cell_worker,
            &TestClass<dim, spacedim>::copier,
            scratch,
            copy,
            assemble_own_cells);
```

Reply:
Erg, I'm sorry, somehow I managed to use a stale version of this code example in my comment. I've fixed this with what I had intended to suggest to you, and have added an extra commit to (hopefully) fix the tester error.